apps/passport-client/src/dispatch.ts

```diff
@@ -448,76 +451,65 @@ async function removePCD(state: AppState, update: ZuUpdate, pcdId: string) {
   update({ pcds: state.pcds });
 }
 
-async function loadFromSync(
+async function loadAfterLogin(
   encryptionKey: string,
-  storage: SyncedEncryptedStorage,
+  storage: StorageWithRevision,
   update: ZuUpdate
 ) {
-  let pcds: PCDCollection;
-  let subscriptions: FeedSubscriptionManager;
-
-  if (isSyncedEncryptedStorageV3(storage)) {
-    pcds = await PCDCollection.deserialize(await getPackages(), storage.pcds);
-    subscriptions = FeedSubscriptionManager.deserialize(
-      new NetworkFeedApi(),
-      storage.subscriptions
-    );
-  } else if (isSyncedEncryptedStorageV2(storage)) {
-    pcds = await PCDCollection.deserialize(await getPackages(), storage.pcds);
-  } else {
-    pcds = await new PCDCollection(await getPackages());
-    await pcds.deserializeAllAndAdd(storage.pcds);
-  }
+  const { pcds, subscriptions } = await deserializeStorage(
+    storage.storage,
+    await getPackages()
+  );
 
   // Poll the latest user stored from the database rather than using the `self` object from e2ee storage.
   const userResponse = await requestUser(
     appConfig.zupassServer,
-    storage.self.uuid
+    storage.storage.self.uuid
   );
   if (!userResponse.success) {
     throw new Error(userResponse.error.errorMessage);
   }
 
-  // assumes that we only have one semaphore identity in Zupass.
+  // TODO: this assumes that we only have one semaphore identity in Zupass.
   const identityPCD = pcds.getPCDsByType(
     SemaphoreIdentityPCDTypeName
   )[0] as SemaphoreIdentityPCD;
 
   let modal: AppState["modal"] = { modalType: "none" };
   if (!identityPCD) {
     // TODO: handle error gracefully
+    // TODO: Also check that identityPCD's commitment matches the one
+    // in storage.self
     throw new Error("no identity found in encrypted storage");
   } else if (
     // If on Zupass legacy login, ask user to set passwrod
     self != null &&
     encryptionKey == null &&
-    storage.self.salt == null
+    storage.storage.self.salt == null
   ) {
     console.log("Asking existing user to set a password");
     modal = { modalType: "upgrade-account-modal" };
   }
 
-  if (subscriptions) {
-    await saveSubscriptions(subscriptions);
-  }
-
   await savePCDs(pcds);
+  await saveSubscriptions(subscriptions);
+  savePersistentSyncStatus({ serverStorageRevision: storage.revision });
   saveEncryptionKey(encryptionKey);
   saveSelf(userResponse.value);
   saveIdentity(identityPCD.claim.identity);
 
   update({
     encryptionKey,
     pcds,
+    serverStorageRevision: storage.revision,
     identity: identityPCD.claim.identity,
     self: userResponse.value,
     modal
   });
 
   await sleep(1);
 
-  console.log("Loaded from sync key, redirecting to home screen...");
-  window.localStorage["savedSyncKey"] = "true";
```

Comment:
Note this line is unrelated to the change at hand, but `window.localStorage["savedSyncKey"]` is vestigial (never read) so I'm removing it to avoid confusion.